tictactoe.js

```diff
@@ -0,0 +1,336 @@
+const readline = require('readline');
+
+const readlineInterface = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout
+});
+
+function ask(questionText) {
+    return new Promise((resolve, reject) => {
+        readlineInterface.question(questionText, resolve);
+    });
+};
+
+let ttTable = ['1', '2', '3', '4', '5', '6', '7', '8', '9'];
+//let whichPlayer = player();
+
+start();
+async function start() {
+
+    console.log('\n'
+        + ' ' + ttTable[0] + ' | ' + ttTable[1] + ' | ' + ttTable[2] + ' \n'
+        + '--- --- ---\n'
+        + ' ' + ttTable[3] + ' | ' + ttTable[4] + ' | ' + ttTable[5] + ' \n'
+        + '--- --- ---\n'
+        + ' ' + ttTable[6] + ' | ' + ttTable[7] + ' | ' + ttTable[8] + ' \n');
+    let turn = await ask("Who's turn first, \n X or O?\n");
+    let xPlayer = await ask('What is your name for playing X? ');
+    let oPlayer = await ask('What is your name for playing O? ');
+    return player(turn, xPlayer, oPlayer);
+
+    async function player(whosTurn, xPlay, oPlay) {
+        console.log('\n'
+            + ' ' + ttTable[0] + ' | ' + ttTable[1] + ' | ' + ttTable[2] + ' \n'
+            + '--- --- ---\n'
+            + ' ' + ttTable[3] + ' | ' + ttTable[4] + ' | ' + ttTable[5] + ' \n'
+            + '--- --- ---\n'
+            + ' ' + ttTable[6] + ' | ' + ttTable[7] + ' | ' + ttTable[8] + ' \n');
+        if (whosTurn === 'X' | whosTurn === 'x') {
+            whosTurn = 'X';
+            let playerX = await ask('\n' + xPlay + " turn for 'X'\nMove to?\n");
+            return move(playerX);
+        } else if (whosTurn === 'O' | whosTurn === 'o') {
+            whosTurn = 'O';
+            let playerO = await ask('\n' + oPlay + " turn for 'O'\n Move to?\n");
+            return move(playerO);
+        } else {
+            console.log("You must enter 'x' or 'o'. ");
+            start();
+        };
+
+        async function move(whichMove) {
```

Comment:
I'm impressed you got this working with all the pops and pushes! 

However, your code could be a lot briefer if you used array indexing (like `ttTable[3]` or `ttTable[whichMove]`) instead. Even though an array can act like a stack, the tictactoe board really is more like an array, since we need to get and set values at random, not just from one end or the other.

If you want to try to rewrite it, it's good you made it a separate function; you can rename this one `xmove` and then add a new `move` and keep the old code around for inspiration until the new function is working.